src/platform/plugins/shared/content_management/server/content_client/content_client_factory.ts

```diff
@@ -60,35 +62,46 @@ export const getMSearchClientFactory =
   ({
     contentRegistry,
     mSearchService,
+    logger,
   }: {
     contentRegistry: ContentRegistry;
     mSearchService: MSearchService;
+    logger: Logger;
   }) =>
   ({
     requestHandlerContext,
+    request,
   }: {
     requestHandlerContext: RequestHandlerContext;
     request: KibanaRequest;
   }) => {
     const msearch = async ({ contentTypes, query }: MSearchIn): Promise<MSearchOut> => {
-      const contentTypesWithStorageContext = contentTypes.map(({ contentTypeId, version }) => {
+      const contentTypesWithStorageContext = contentTypes.reduce<
+        Array<{ contentTypeId: string; ctx: StorageContext }>
+      >((acc, { contentTypeId, version }) => {
         const contentDefinition = contentRegistry.getDefinition(contentTypeId);
 
-        const storageContext = getStorageContext({
-          contentTypeId,
-          version: version ?? contentDefinition.version.latest,
-          ctx: {
-            contentRegistry,
-            requestHandlerContext,
-            getTransformsFactory: getServiceObjectTransformFactory,
-          },
-        });
+        if (contentDefinition.storage.mSearch) {
+          const storageContext = getStorageContext({
+            request,
+            contentTypeId,
+            version: version ?? contentDefinition.version.latest,
+            ctx: {
+              contentRegistry,
+              requestHandlerContext,
+              getTransformsFactory: getServiceObjectTransformFactory,
+            },
+          });
+          acc.push({
+            contentTypeId,
+            ctx: storageContext,
+          });
+        } else {
+          logger.warn(`mSearch method missing for content type "${contentTypeId}" v${version}.`);
+        }
 
-        return {
-          contentTypeId,
-          ctx: storageContext,
-        };
-      });
+        return acc;
+      }, []);
```

Comment:
The `mSearch` method is not always available but the previous code assumed it to be.

In most cases, the types used by `mSearch`, like from the **Visualize library**, are always defined but if another vis wants to provide a type alias and does not specify `mSearch`, it will break the **Visualize library**.

This collects all the content type factories like before but ignores them if they have not defined `mSearch`. Logs a server warning if requested type is missing `mSearch`.